src/components/TimerContainer.js

```diff
@@ -0,0 +1,42 @@
+import React, { useEffect, useState } from 'react';
+import { publish, subscribe } from './Event';
+
+function TimerContainer(props) {
+  const { questionNum } = props;
+  const [count, setCount] = useState(20);
+
+  const updateCount = () => {
+    if (count > 1) {
+      setCount((prevState) => prevState - 1);
+    } else if (questionNum === 20) {
+      setCount(0);
+      publish('NextQuestion');
+    } else {
+      publish('NextQuestion');
+    }
+  };
+  useEffect(() => {
+    const timer = setInterval(updateCount, 1000);
+    subscribe('ResetInterval', (data) => {
```

Comment:
Shouldn't this be called in update count ?

Reply:
Done